models/articles.models.js

```diff
@@ -11,8 +11,18 @@ exports.selectArticlesById = (article_id) => {
     });
 };
 
-exports.selectAllArticles = () => {
-    
+exports.selectAllArticles = (topic) => {
+  
+  if(topic) {
+    return db.query(`SELECT * FROM articles WHERE topic = $1`, [topic])
+    .then((result) => {
+      if(result.rows.length === 0){
+        return Promise.reject({status: 404, msg: 'article does not exist for matched query'})
+      }
+      return result.rows;
+    })
+  }
```

Comment:
Wit this query, you are asking SQL for all info of articles. This also includes body which shouldn't be included. Also there is no comment count for these. Please have a go at refactoring this model to make use of just one db.query. The notes on complex queries will really help with this 😄